roles/container-engine/docker/defaults/main.yml

```diff
@@ -13,10 +13,6 @@ docker_repo_info:
 
 docker_cgroup_driver: systemd
 
-docker_dns_servers_strict: true
```

Comment:
```
$ git grep docker_dns_servers_strict
inventory/sample/group_vars/all/docker.yml:docker_dns_servers_strict: false
roles/container-engine/docker/defaults/main.yml:docker_dns_servers_strict: true
roles/container-engine/docker/tasks/set_facts_dns.yml:    msg: "Too many nameservers. You can relax this check by set docker_dns_servers_strict=false in all.yml and we will only use the first 3."
roles/container-engine/docker/tasks/set_facts_dns.yml:  when: docker_dns_servers|length > 3 and docker_dns_servers_strict|bool
roles/container-engine/docker/tasks/set_facts_dns.yml:  when: docker_dns_servers|length > 3 and not docker_dns_servers_strict|bool
roles/kubespray-defaults/defaults/main.yaml:docker_dns_servers_strict: false
```

Aren't you changing the default by deleting this line ?

Reply:
Also should the default always go to `kubespray-defaults` even if only used in 1 role ? It's cleaner to shrink `kubespray-defaults` WDYF

Reply:
Yup I didn't wanted to change this too much but I find it weird to have docker default in kubespray-default and not docker default..

Reply:
And regarding `docker_dns_servers_strict` I'm not sure at all what to do with this as there was true/false in different defaults/sample.

I'm more with setting it to false and not trimming, and letting user opt-in to true if they want to, but up to you/reviewers

Reply:
Just update "Does this PR introduce a user-facing change?" and I'm good
(not using docker myself :) )